### PR TITLE
Detach disabling "talk" speech mode warning dialog

### PR DESCRIPTION
**Pull request by hwf1324**
### Link to issue number:

Related #15894

### Summary of the issue:

Disable the "talk" speech mode warning dialog box can be completely separated for judgment, as the status of this option does not need to be guaranteed when determining the settings

### Description of user facing changes

When "talk" is deselected, a warning dialog box pops up. If "no" is chosen, then "talk" is set back to the selected state

### Description of development approach

Move the "talk" speech mode warning dialog to the method bound to the `wx.EVT_CHECKLISTBOX` event.

### Testing strategy:

1. Open the speech settings panel
2. Deselect the "talk" speech mode in loop speech mode

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

### CODE REVIEW

**Comment by hwf1324** [comment]
This PR does not require a change log

**Comment by lukaszgo1** [comment]
Given discussion in #15894 especially https://github.com/nvaccess/nvda/issues/15894#issuecomment-1848020068 this was quite unpopular with users. If you believe that these arguments are not valid please explain why. Otherwise I suggest to close this.

**Comment by hwf1324** [comment]
> Given discussion in #15894 especially [#15894 (comment)](https://github.com/nvaccess/nvda/issues/15894#issuecomment-1848020068) this was quite unpopular with users. If you believe that these arguments are not valid please explain why. Otherwise I suggest to close this.

No, this will not trigger warning dialog boxes frequently.
This will only trigger the warning dialog when the "talk" speech mode is canceled.
The judgment for selecting two options is still made before saving.
I believe users will not frequently cancel/select an option

**Comment by lukaszgo1** [comment]
I believe we should be aiming for consistency here, i.e. all warnings appear at the same time that is when closing the dialog. Lets wait for more feedback on this PR.

**Comment by CyrilleB79** [comment]
On the contrary, I believe that we should issue warnings / error as soon as possible.

It was not possible for the "select at least two items" use case, since a standard way to select two items may be transitioning through a state where only one item is selected.

On the contrary, as soon as a user deselects "Talk":
* either it was unintended, in which case the warning is more than useful
* or it's intended, but we have already agreed that the use case is so rare that the warning is acceptable in this case

In conclusion, I am in favor of the changes in this PR.

**Comment by seanbudd** [comment]
if we don't want users to deselect talk I think we should remove it from the user accessible options entirely and forcibly include it internally. I think we initially included it in the settings as an earlier goal was to ensure either "on-demand" or "talk" was included, not just "talk".

File: source/gui/settingsDialogs.py [hunk 1]
```diff
@@ -1686,6 +1687,28 @@ def onSave(self):
 			mIndex for mIndex in range(len(self._allSpeechModes)) if mIndex not in self.speechModesList.CheckedItems
 		]
 
+	def onSpeechModesListChange(self, evt):
```
[the remark this answers is not in the document]
Comment (hwf1324) [reply source/gui/settingsDialogs.py:1690]: 好的，但是对于绑定到`wx.EVT*`的事件处理程序，是否应该明确何时应该标记为私有？

[the remark this answers is not in the document]
Comment (seanbudd) [reply source/gui/settingsDialogs.py]: Functions should be marked as private if there is no reason for add-ons to use it, or the API is unstable (i.e. possibly going to be changed in future releases).

[the remark this answers is not in the document]
Comment (hwf1324) [reply source/gui/settingsDialogs.py]: Thanks, got it, depending on the decision this function may be removed.

**Comment by lukaszgo1** [comment]
> if we don't want users to deselect talk I think we should remove it from the user accessible options entirely and forcibly include it internally. I think we initially included it in the settings as an earlier goal was to ensure either "on-demand" or "talk" was included, not just "talk".


From the discussion on #15806 and #15873 the goal was to allow users to deselect 'talk', since this may be potentially useful for magnification users, but make sure this is not done by mistake. Frankly given how much time has been spent discussing if it should be de-selectable or not I'd suggest not to spent more time on this i.e. merge this PR, as it improves the behavior. IMO the UX can be reconsidered if one of the following happens:
1. We receive complains from users who disabled 'talk' mode by mistake despite the warning or
2. We receive a lot of reports about warning being annoying from people who intentionally want to disable 'talk' mode.

**Comment by seanbudd** [comment]
Thanks for the reminder and explainer @lukaszgo1

**Review by seanbudd — APPROVED** [review]
